### PR TITLE
Fix button border color of LeaveSpaceDialog

### PR DESCRIPTION
**Pull request by luixxiul**
Fixes https://github.com/vector-im/element-web/issues/21365

This commit fixes the confirmation button border color of LeaveSpaceDialog by moving `.mx_LeaveSpaceDialog` inside `.mx_Dialog`. Thanks to this change `!important` and `margin-top` became redundant.

Signed-off-by: Suguru Hirahara <luixxiul@users.noreply.github.com>

after

Notes: Fix button border color of LeaveSpaceDialog

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Fix button border color of LeaveSpaceDialog ([\#8010](https://github.com/matrix-org/matrix-react-sdk/pull/8010)). Fixes vector-im/element-web#21365. Contributed by @luixxiul.

Preview: https://pr8010--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: res/css/views/dialogs/_LeaveSpaceDialog.scss [hunk 1]
```diff
@@ -19,58 +19,56 @@ limitations under the License.
...
+                    }
+                }
 
-        > p {
-            color: $primary-content;
-        }
-    }
-
-    .mx_Dialog_buttons {
-        margin-top: 20px;
```
Comment (luixxiul) [inline res/css/views/dialogs/_LeaveSpaceDialog.scss:69]: Removed based on this: https://github.com/matrix-org/matrix-react-sdk/blob/2ab8b46ba3fcac15f66b0c040d7a822aa9ae882e/res/css/_common.scss#L411

File: res/css/views/dialogs/_LeaveSpaceDialog.scss [hunk 2]
```diff
@@ -19,58 +19,56 @@ limitations under the License.
...
-    }
-}
 
-.mx_LeaveSpaceDialog {
-    width: 440px;
-    display: flex;
-    flex-direction: column;
-    flex-wrap: nowrap;
-    height: 520px;
+        .mx_LeaveSpaceDialog {
```
Comment (luixxiul) [inline res/css/views/dialogs/_LeaveSpaceDialog.scss:23]: Moved inside `.mx_Dialog` here.

File: src/components/views/dialogs/LeaveSpaceDialog.tsx [hunk 3]
```diff
@@ -105,6 +105,7 @@ const LeaveSpaceDialog: React.FC<IProps> = ({ space, onFinished }) => {
         </div>
         <DialogButtons
             primaryButton={_t("Leave space")}
+            primaryButtonClass="danger"
```
Comment (luixxiul) [inline src/components/views/dialogs/LeaveSpaceDialog.tsx:108]: Note: This makes styling with `.mx_Dialog_buttons` unnecessary.

**Review by t3chguy — APPROVED** [review]
Looks sane to me, thanks!

**Comment by luixxiul** [comment]
Please let me know if it breaks something, thanks!

**Pull request merged**